libraries/chain/db_block.cpp

```diff
@@ -131,8 +131,19 @@ bool database::_push_block(const signed_block& new_block)
    uint32_t skip = get_node_properties().skip_flags;
    if( !(skip&skip_fork_db) )
    {
-      /// TODO: if the block is greater than the head block and before the next maitenance interval
-      // verify that the block signer is in the current set of active witnesses.
+
+      // If the block is greater than the head block and before the next maintenance interval
+      if (new_block.block_num() > head_block_num()
+            && new_block.timestamp <  get<dynamic_global_property_object>(dynamic_global_property_id_type()).next_maintenance_time )
+      {
+         // make sure the block signer is in the current set of active witnesses
+         //TODO: Do we need to validate signature, or has this already been done?
+         auto witnesses = get_global_properties().active_witnesses;
+         if (std::find(witnesses.begin(), witnesses.end(), new_block.witness ) == witnesses.end())
```

Comment:
`active_witnesses` is a `flat_set`, I think it's clearer and more efficient to find directly with `witnesses.find(...)`.

Reply:
Changed to use witnesses.find